server/routes/index.ts

```diff
@@ -745,20 +745,39 @@ export function defineRoutes(router: IRouter) {
   router.delete(
     {
       path: `${API_PREFIX}/configuration/cache`,
-      validate: false,
+      validate: {
+        body: schema.object({
+          dataSourceId: schema.maybe(schema.string()),
+        }),
+      },
     },
     async (context, request, response) => {
-      const client = context.security_plugin.esClient.asScoped(request);
-      let esResponse;
-      try {
-        esResponse = await client.callAsCurrentUser('opensearch_security.clearCache');
-        return response.ok({
-          body: {
-            message: esResponse.message,
-          },
-        });
-      } catch (error) {
-        return errorResponse(response, error);
+      if (!dataSourceEnabled || !request.body?.dataSourceId) {
+        const client = context.security_plugin.esClient.asScoped(request);
+        let esResponse;
+        try {
+          esResponse = await client.callAsCurrentUser('opensearch_security.clearCache');
+          return response.ok({
+            body: {
+              message: esResponse.message,
+            },
+          });
+        } catch (error) {
+          return errorResponse(response, error);
+        }
+      } else {
+        const client = context.dataSource.opensearch.legacy.getClient(request.body?.dataSourceId);
```

Comment:
this client will be setup by opensearch-core? how do we test this? Where can we set the boundary to say that this is out of scope of testing in security-dashboards-plugin?

Reply:
In this case, we need to add unit test here to test the business logic, and when condition is met, this getClient method is called. In addition, we can add cypress test to test end to end.

Reply:
Added tests in: [88fa112](https://github.com/opensearch-project/security-dashboards-plugin/pull/1818/commits/88fa11278e4b6ef938390e9c25ed2de02665cc33), this can be simplified once the TLS issue in core is resolved via: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6171, but it should work for now.